recipes/cose-c/all/conanfile.py

```diff
@@ -1,92 +1,100 @@
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, 11)
 
     def requirements(self):
-        self.requires("cn-cbor/1.0.0")
+        self.requires("cn-cbor/1.0.0", transitive_headers=True)
 
         if self.options.with_ssl == "mbedtls":
-            self.requires("mbedtls/2.23.0")
+            self.requires("mbedtls/2.16.12", transitive_headers=True)
         else:
-            self.requires("openssl/1.1.1h")
+            self.requires("openssl/[>=1.1 <4]", transitive_headers=True)
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        os.rename(glob.glob("COSE-C-*")[0], self._source_subfolder)
-
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["COSE_C_COVERALLS"] = False
-        self._cmake.definitions["COSE_C_BUILD_TESTS"] = False
-        self._cmake.definitions["COSE_C_BUILD_DOCS"] = False
-        self._cmake.definitions["COSE_C_BUILD_DUMPER"] = False
-        self._cmake.definitions["COSE_C_USE_MBEDTLS"] = self.options.with_ssl == "mbedtls"
-        self._cmake.definitions["COSE_C_USE_FIND_PACKAGE"] = True
-        self._cmake.definitions["COSE_C_EXPORT_TARGETS"] = True
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True
+        tc.cache_variables["COSE_C_COVERALLS"] = False
+        tc.cache_variables["COSE_C_BUILD_TESTS"] = False
+        tc.cache_variables["COSE_C_BUILD_DOCS"] = False
+        tc.cache_variables["COSE_C_BUILD_DUMPER"] = False
+        tc.cache_variables["COSE_C_USE_MBEDTLS"] = self.options.with_ssl == "mbedtls"
+        tc.cache_variables["COSE_C_USE_FIND_PACKAGE"] = True
+        tc.cache_variables["COSE_C_EXPORT_TARGETS"] = True
+        tc.generate()
+        deps = CMakeDeps(self)
+        deps.set_property("openssl", "cmake_file_name", "OPENSSL")
+        deps.set_property("mbedtls", "cmake_target_name", "mbedtls")
+        deps.generate()
+
+    def _patch_sources(self):
+        # For ${OPENSSL_LIBRARIES} and ${OPENSSL_INCLUDE_DIR}
+        replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"),
+                        "OpenSSL", "OPENSSL")
```

Comment:
You shouldn't need this workaround, OPENSSL_LIBRARIES and OPENSSL_INCLUDE_DIR are defined as extra vars in CMakeDeps of openssl